pipelines/scripts/util.py

```diff
@@ -11,9 +14,59 @@ def run_az_cli_command(cls, args: List[str]):
         """
             Run an az cli command. Raises a `RuntimeException` if something goes wrong
         """
-        print(f"Running command: '{' '.join(args)}'")
+        logging.info(f"Running command {' '.join(args)}")
         try:
             return subprocess.check_output(args)
         except subprocess.CalledProcessError as e:
             exception = e
```

Comment:
Might be helpful to maybe show the command output/error message explicitly or log the full AZ CLI command that failed to debug ? Might be easier ? Just a thought

Reply:
I want to explicitly show the command just for debugging purposes in the future, it's helpful to see what command was run in ADO if troubleshooting something going wrong

Reply:
Completely agree you're already doing part of this with this command. 

I suggest maybe this improvement to it : 

try:
    return subprocess.check_output(args, stderr=subprocess.STDOUT)
except subprocess.CalledProcessError as e:
    command_str = ' '.join(args)
    raise RuntimeError(f"Error running command: {command_str}\nOutput:\n{e.output.decode('utf-8')}")
    
    
    So you get the command that was run, output produced and why it failed

Reply:
I'd prefer to use the current approach, i want to always log the command just for visibility since debugging ADO can be tricky sometimes, the full trace can be helpful incase there is a transient error/mistake in the command